app/react/Viewer/components/OCRDisplay.tsx

```diff
@@ -0,0 +1,190 @@
+import React, { useEffect, useState } from 'react';
+import { connect, ConnectedProps } from 'react-redux';
+import { FileType } from 'shared/types/fileType';
+import { Translate } from 'app/I18N';
+import socket from 'app/socket';
+import { bindActionCreators } from 'redux';
+import { postToOcr, getOcrStatus } from '../actions/ocrActions';
+import { ocrDisplayTips } from '../utils/ocrDisplayTips';
+import { reloadDocument } from '../actions/documentActions';
+
+type OCRDisplayProps = {
+  file: FileType;
+};
+
+const formatDate = (time: number, locale: string) => {
+  const date = new Date(time);
+  return date.toLocaleString(locale);
+};
+
+const mapStateToProps = ({ settings, locale }: any) => {
+  const toggleOCRButton = settings.collection.get('toggleOCRButton');
+  return {
+    ocrIsToggled: toggleOCRButton || false,
+    locale,
+  };
+};
+
+const mapDispatchToProps = (dispatch: any) =>
+  bindActionCreators({ loadDocument: reloadDocument }, dispatch);
+
+const connector = connect(mapStateToProps, mapDispatchToProps);
+type mappedProps = ConnectedProps<typeof connector>;
+type ComponentProps = OCRDisplayProps & mappedProps;
+
+const OCRDisplay = ({ file, ocrIsToggled, locale, loadDocument }: ComponentProps) => {
```

Comment:
I think this component is wrongly named.  OCRDisplay sounds more like the actual result of the OCR.  This is more like OCRStatus? This includes the name of the file.